app/models/game.rb

```diff
@@ -30,27 +39,60 @@ def populate_game!
     save
   end
 
-  def check?
-    white_king = pieces.find_by(type: 'King', color: 'white')
-    black_pieces = pieces.where(color: 'black')
+  # Keep track of current king.  We don't care what the other king is doing.
+  def check?(color)
+    king = pieces.find_by(type: 'King', color: color)
+    opposing_pieces = pieces.where(color: king.opposite_color, captured: false)
 
-    black_pieces.each { |black_piece| return true if black_piece.valid_move?(white_king.x_coordinate, white_king.y_coordinate) }
+    opposing_pieces.each { |opposing_piece| return true if opposing_piece.valid_move?(king.x_coordinate, king.y_coordinate) }
+    false
+  end
 
-    black_king = pieces.find_by(type: 'King', color: 'black')
-    white_pieces = pieces.where(color: 'white')
+  def checkmate?(color)
+    #  a. king has to be in check
+    return false unless check?(color)
 
-    white_pieces.each { |white_piece| return true if white_piece.valid_move?(black_king.x_coordinate, black_king.y_coordinate) }
+    #  a1. king cannot get out of check (has no more valid moves to get out of check)
+    #  a2. king cannot be blocked by another piece (any color)
+    #  a3. pieces checking king check cannot be captured
 
-    false
+    # Go through every possible move for every single piece I have and see if you can get out of checkmate
+    pieces.find_by(color: color, captured: false).each do |piece|
```

Comment:
find_by worked fine before, but if you want to change it to where, that is fine too.  
I just pulled your changes locally and it looks like it's already done.
One question though, why use `.find_each` Isn't it a bit excessive?   It's used for batch processing only.

http://apidock.com/rails/ActiveRecord/Batches/ClassMethods/find_each

"Note: This method is only intended to use for batch processing of large amounts of records that wouldn’t fit in memory all at once. If you just need to loop over less than 1000 records, it’s probably better just to use the regular find methods."

As far as I'm concerned, it passes, I'm good with merging it into master.  We should move on.  I know stalemate needs some of the code changes for the moves.